agent/agent.go

```diff
@@ -398,15 +408,42 @@ func (a *Agent) newExecutionProviderParams(req *agentapi.DeployRequest, tmpFile
 	return params, nil
 }
 
+func (a *Agent) setupSignalHandlers() {
```

Comment:
Similar concern regarding use of `os.Exit` here in `setupSignalHandlers`.

Reply:
https://pkg.go.dev/runtime#Goexit

Reply:
Not sure we need `Goexit` here either, although this is interesting.